### PR TITLE
std.internal.math.gammafunction: loosen some tests for 64-bit reals.

### PR DESCRIPTION
**Pull request by kinke**
As supplement for https://github.com/JohanEngelen/phobos/commit/2ac2581fe49da475bf6f687cfb7bcb9c9ddf8b71 (1st commit of https://github.com/D-Programming-Language/phobos/pull/3045; that commit alone doesn't depend on the other PR).

### CODE REVIEW

**Comment by WalterBright** [comment]
Why? I like to have them tight, because they let me know when the algorithm changes in a way that makes the result less accurate.

**Comment by kinke** [comment]
Simple - these tests assume 80-bit reals, and it's time to support other targets too, don't you think? We're taking about 3 relaxed assertions and 9 coverage tests now only enabled for x87 reals...

**Comment by WalterBright** [comment]
Use `version` blocks for tests for targets that have reduced precision. The tests for all platforms should not be the worst precision on any platform. Using max precision for each target is necessary so we know when that precision decays.

**Comment by kinke** [comment]
Summary:
* line 526: previous input results in overflow for 64-bit real => extended by a dedicated test
* line 902: allow 14 instead of only 13 least-significant mantissa bits to diverge so that no additional modification is required for 64-bit reals
* disable existing extended coverage tests for non-80-bit reals as, iirc, most of those yield NaN with 64-bit reals and the coverage has most likely been measured for the x87 inline asm code only
* line 1421: relax the upper limit for the absolute difference; the new limit is used in all other asserts of that test

Using version blocks results in way clumsier code, we have enough of those PPC, PPC64, ARM, ARM64, MIPS, MIPS64 etc. stuff already in there, so I'll stick with the `static if` to check the precision of the `real` type.

File: std/internal/math/gammafunction.d [hunk 1]
```diff
@@ -1418,7 +1423,7 @@ assert(gammaIncompleteComplInv(3, 0)==real.infinity);
 // Fixed a bug that caused gammaIncompleteCompl to return a wrong value when
 // x was larger than a, but not by much, and both were large:
 // The value is from WolframAlpha (Gamma[100000, 100001, inf] / Gamma[100000])
-assert(fabs(gammaIncompleteCompl(100000, 100001) - 0.49831792109) < 0.000000000005);
+assert(fabs(gammaIncompleteCompl(100000, 100001) - 0.49831792109) < 0.00000005);
```
Comment (WalterBright) [inline std/internal/math/gammafunction.d]: Again, please do not reduce the precision of 80 bit tests. Add a separate test for 64 bit reals.

File: std/internal/math/gammafunction.d [hunk 2]
```diff
@@ -899,7 +902,7 @@ unittest { // also tested by the normal distribution
     // These arbitrary points are chosen to give good code coverage.
     assert(feqrel(betaIncomplete(8, 10, 0.2), 0.010_934_315_234_099_2L) >=  real.mant_dig - 5);
     assert(feqrel(betaIncomplete(2, 2.5, 0.9),0.989_722_597_604_452_767_171_003_59L) >= real.mant_dig - 1 );
-    assert(feqrel(betaIncomplete(1000, 800, 0.5), 1.179140859734704555102808541457164E-06L) >= real.mant_dig - 13 );
+    assert(feqrel(betaIncomplete(1000, 800, 0.5), 1.179140859734704555102808541457164E-06L) >= real.mant_dig - 14 );
```
Comment (WalterBright) [inline std/internal/math/gammafunction.d]: Why the change?

**Comment by kinke** [comment]
Done.

**Comment by WalterBright** [comment]
thanks!

**Comment by WalterBright** [comment]
Auto-merge toggled on
